API/things-api/app.js

```diff
@@ -71,18 +84,181 @@ app.get('/view', function(req, res){
+****************************************************/
+app.get('/tagitem/:id/:tag', function(req, res){
+  //first query the database
+  //then return the results to the user
+
+      pool.connect(function(err, client, done) {
+        if(err) {
+            return console.error('error fetching client from pool', err);
+        }
+        client.query('INSERT INTO tags VALUES ($1, $2)',
+                    [req.params.tag, req.params.id], function(err, result) {
+            //call `done()` to release the client back to the pool
+            done();
+            errResultHandler(err, 'Tag Added Successfully', res);
+        });
+    });
+});
+
+
+
+/****************************************************
+* /path     /shoppinglist
+* /params   null
+* /brief    Returns all items with quantity less than threshold
+*
+* /author   Luke
+****************************************************/
+app.get('/shoppinglist', function(req, res){
+  //first query the database
+  //then return the results to the user
+
+      pool.connect(function(err, client, done) {
+        if(err) {
+            return console.error('error fetching client from pool', err);
+        }
+        client.query('SELECT item_name AS name, description, price FROM items WHERE quantity < threshold', [], function(err, result) {
+            //call `done()` to release the client back to the pool
+            done();
+            errResultHandler(err, result.rows, res);
+        });
+    });
+});
+
+
+
+/****************************************************
+* /path     /stats/:id
+* /params
+*
+*
+* /brief    Route to pull satistics for an item
+*           should probably be renamed to be more specific
+*
+* /author   Andrew McCann
+* /date     2/3/2017
+****************************************************/
+app.get('/stats/:id', function(err, client, done) {
+    pool.connect(function(err, client, done) {
+        if(err) {
+            return console.error('Error fetching client from pool', err);
+        }
+        /*
+        client.query('INSERT INTO transactions(item_id, person, qty_changed) VALUES ($1, $2, $3)', [id, person, qty], function(err, result) {
+        client.query('SELECT item_name, quantity, blahbalh
+        client.query('SELECT * FROM transactions WHERE item_name = id', [], function(err, result) {
+            done();
+
+            if(err)
+                return console.error('Error returned from DB', err);
+
+            res.jsonp(result.rows);
+        }*/
+    });
+});
+
+
+
+/****************************************************
+* /path     /stats/range/:start_date/:end_date
+* /params   :start_date - Beginning of time frame
+*           :end_date - End of time frame
+*
+* /brief    Route to pull all stats from a date_range
+*
+* /author   Andrew McCann
+* /date     2/3/2017
+****************************************************/
+app.get('/stats/range/:start_date/:end_date', function(err, client, done) {
+    pool.connect(function(err, client, done) {
```

Comment:
This route is unfinshed, and dangerous!!!!   route checks out a client from pool, but never gives it back.